src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.SetThreadDpiHostingBehavior.cs

```
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Windows.Forms;
+
+namespace Windows.Win32
+{
+    internal static partial class PInvoke
+    {
+        /// <summary>
+        ///  Sets the thread DPI hosting behavior.
+        /// </summary>
+        /// <returns>
+        ///  The old thread DPI hosting behavior if the API is available in this version of OS.
+        ///  Otherwise, <see cref="DPI_HOSTING_BEHAVIOR.DPI_HOSTING_BEHAVIOR_INVALID"/>.
+        /// </returns>
+        public static DPI_HOSTING_BEHAVIOR SetThreadDpiHostingBehaviorInternal(DPI_HOSTING_BEHAVIOR dpiHostingBehavior)
+        {
+            if (dpiHostingBehavior != DPI_HOSTING_BEHAVIOR.DPI_HOSTING_BEHAVIOR_MIXED)
+            {
+                // Winforms only affects the behavior of thread hosting in mixed DPI mode situations, and does not apply to any other scenarios.
+                throw new ArgumentException(nameof(dpiHostingBehavior), dpiHostingBehavior.ToString());
+            }
+
+            if (!OsVersion.IsWindows10_18030rGreater())
+            {
+                throw new PlatformNotSupportedException();
```

Are we throwing in other places? I thought we wanted to fail silently.

---

We were catching `EntryPointNotFoundException` before.

---

There is no scenario at this point that tries to set behavior that is not mixed and calling this explicitly on legacy OS and running PermonV2.  Calling into Win32 knowingly it is not available isn't helpful.

---

We can just skip the call when it's no available, the app is SA or UA anyway.

---

Sorry i wasn't right. Looking changes in Github browsers didn't give clear picture.  This method isn't used currently and is not needed any more after refactored code addressing your original review.  We are returning silently when running on legacy OS and never set hosting behavior or call it.